sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsHostMemoryStore.scala

```diff
@@ -86,13 +95,18 @@ class RapidsHostMemoryStore(
           hostBuffer.close()
           throw e
       }
+      val spillPriority = other.getSpillPriority - (allocationMode match {
+        case Pinned => 200
+        case Pooled => 100
+        case Direct => 0
```

Comment:
Would be good to have some comments here or maybe make this a bit more formal.  The problem is that the application has no idea that this adjustment is being made, and the values are hidden and arbitrary.  If the application uses very low values for priority then instead of making the spill priority primarily controlled by the application and subtly controlled by host buffer location it will be dominated by host buffer location.

Wondering if we should officially partition the spill priority value to make room for store-specific priorities and then discuss whether those priorities should dominate or whether app priorities dominate.

Reply:
Created constants in `SpillPriorities` to better document them. Looks like the existing priorities are mostly at extreme ends of `Long`.